crates/type-system/src/lib.rs

```diff
@@ -0,0 +1,11 @@
+#![feature(lint_reasons)]
```

Comment:
Do we want to pull in the same lints as for the HASH backend?

Reply:
Good idea, have brought most of them in, we can remove if they cause problems